### PR TITLE
Core 430 add ensure auth to smart agent core

### PR DESCRIPTION
**Pull request by karladler**

### CODE REVIEW

**Comment by wulfraem** [comment]
- updated test to use `api-blockchain-core` to build auth header
- moved contents from common folder to lib folder
- resolve merge conflicts after https://github.com/evannetwork/edge-server-seed/pull/2

File: middlewares/authentication.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+'use strict'
+const { api } = require('actionhero')
+
+const MAX_AGE = 1000 * 60 * 5 // max age of signed message is 5m
+
+/**
+ * verify auth headers; example for creating a signed auth header:
+ * web3.eth.accounts.sign(`${accountId}|${Date.now()}`, `0x${privateKey}`)
```
Comment (S3bb1) [inline middlewares/authentication.js]: Should we adjust the comment? :D
```suggestion
 * web3.eth.accounts.sign(`${Date.now()}`, `0x${privateKey}`)
```

File: actions/authenticated.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (S3bb1) [inline actions/authenticated.js:1]: why this action is needed?

Comment (karladler) [reply actions/authenticated.js:1]: just as an example how to use authenticated middleware and for unit tests.

**Pull request merged**